core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -258,18 +265,46 @@ private[spark] class SecurityManager(
     isUserInACL(user, modifyAcls, modifyAclsGroups)
   }
 
+  /**
+   * Check to see if authentication is enabled and SSL for RPC is disabled (as they are mutually
+   * exclusive)
+   * @return Whether authentication is enabled and SSL for RPC is disabled
+   */
+  private def isAuthenticationEnabledAndSslRpcDisabled(): Boolean = {
```

Comment:
Why is authentication mutually exclusive from rpc ssl ?
For example, external shuffle service uses it to validate identity of the incoming request.

IMO let us keep SSL for securing over-the-wire comm

Reply:
Thanks! I had misread the docs earlier and thought authentication required encryption, but re-reading now I realize it's the other way around. I can adjust. While TLS can handle authentication too (with the right trust stores) it makes sense to keep existing auth mechanisms working over TLS.

Just to confirm, we do want to keep the legacy encryption and TLS mutually exclusive right? I don't see a benefit to double encrypting (just overhead)

Reply:
Agree, does not make sense to do existing encryption over TLS :-)